samples/nrf9160/gps/src/main.c

```diff
@@ -0,0 +1,255 @@
+#include<zephyr.h>
+#include<bsd.h>
+#include<nrf_socket.h>
+#include<net/socket.h>
+#include<stdio.h>
+
+#define AT_XSYSTEMMODE "AT\%XSYSTEMMODE=0,0,1,0"
+#define AT_MAGPIO      "AT\%XMAGPIO=1,0,0,1,1,1574,1577"
+#define AT_CFUN        "AT+CFUN=1"
+
+static const char update_indicator[] = {'\\', '|', '/', '-'};
+static const char at_commands[][31]  = { AT_XSYSTEMMODE, AT_MAGPIO, AT_CFUN };
+static int        fd;
+static u64_t      fix_timestamp;
+
+void bsd_recoverable_error_handler(uint32_t error)
+{
+	printk("Err: %lu\n", error);
+}
+
+void bsd_irrecoverable_error_handler(uint32_t error)
+{
+	printk("Irrecoverable: %lu\n", error);
+}
+
+static int enable_gps(void)
+{
+	int  at_sock;
+	int  bytes_sent;
+	int  bytes_received;
+	char buf[3] = {0};
+
+	at_sock = socket(AF_LTE, 0, NPROTO_AT);
+	if (at_sock < 0) {
+		return -1;
+	}
+
+	for (int i = 0; i < 3; i++) {
+		bytes_sent = send(at_sock, at_commands[i],
+				  strlen(at_commands[i]), 0);
+
+		if (bytes_sent < 0) {
+			return -1;
+		}
+
+		do {
+			bytes_received = recv(at_sock, buf, 2, 0);
+		} while (bytes_received == 0);
+
+		if (strstr(buf, "OK") == NULL) {
```

Comment:
Since you always read only two bytes, maybe `memcmp` would be better?

Reply:
ok